annotator-core/src/main/java/edu/ucr/cs/riple/core/metadata/trackers/ParameterRegionTracker.java

```diff
@@ -62,7 +62,19 @@ public Optional<Set<Region>> getRegions(Location location) {
             .collect(Collectors.toSet());
     // Add the method the fix is targeting.
     regions.add(new Region(parameter.clazz, parameter.method));
-    // Add all call sites.
+    // Add all call sites. It will also reserve call sites to prevent callers pass @Nullable
+    // simultaneously while investigating parameters impact.
+    // See example below:
+    // void foo(Object o) {
+    //   bar(o);
+    // }
+    // void bar(Object o)
+    //
+    // We need to make sure that while investigating impact of `@Nullable` on bar#o, other callers
+    // are not passing `@Nullable` to bar#o. Since the corresponding error will not be triggered
```

Comment:
To be clear, what we want to make sure is that while investigating the impact of adding `@Nullable` at `bar#o`, we aren't also simultaneously trying to get the impact of making some value that gets passed to `bar#o` `@Nullable`. Or, at least, that we are not trying to calculate the impact of those two changes as _independent_ changes, yet using the same NullAway run to count the errors.

So, isn't this just saying that the regions of a method argument include the method body itself and also the method bodies of all its callers? Since errors reported on that argument can change with changes affecting nullability at either it's caller (actual argument) or callee (formal argument).

Presumably, similar logic holds for method returns.

Is that all that this comment is saying? Or am I still missing what these tests are about?

Reply:
Yes exactly, but we aren't trying to use the same NullAway run to count the errors, we are actually forcing them to investigate the impacts in two different runs of NullAway.